akka-docs/rst/scala/persistence-query.rst

```diff
@@ -0,0 +1,212 @@
+``NoRefresh`` hint to the query:
+
+.. includecode:: code/docs/persistence/query/PersistenceQueryDocSpec.scala#all-persistence-ids-snap
+
+``EventsByPersistenceId`` is a query equivalent to replaying an :ref:`PersistentActor <event-sourcing>`,
+however, since it is a stream it is possible to keep it alive and watch for additional incoming events persisted by the
+persistent actor identified by the given ``persistenceId``. Most journal will have to revert to polling in order to achieve
+this, which can be configured using the ``RefreshInterval`` query hint:
+
+.. includecode:: code/docs/persistence/query/PersistenceQueryDocSpec.scala#events-by-persistent-id-refresh
+
+``EventsByTag`` allows querying events regardles of which ``persistenceId`` they are associated with. This query is hard to
+implement in some journals or may need some additional preparation of the used data store to be executed efficiently,
+please refer to your read journal plugin's documentation to find out if and how it is supported. The goal of this query
+is to allow querying for all events which are "tagged" with a specific tag - again, how exactly this is implemented
+depends on the used journal.
+
+.. note::
+  A very important thing to keep in mind when using queries spanning multiple persistenceIds, such as ``EventsByTag``
+  is that the order of events at which the events appear in the stream rarely is guaranteed (or stable between materializations).
+
+  Journals *may* choose to opt for strict ordering of the events, and should then document explicitly what kind of ordering
+  guarantee they provide - for example "*ordered by timestamp ascending, independently of persistenceId*" is easy to achieve
+  on relational databases, yet may be hard to implement efficiently on plain key-value datastores.
+
+In the example below we query all events which have been tagged (we assume this was performed by the write-side using an
+:ref:`EventAdapter <event-adapter-scala>`, or that the journal is smart enough that it can figure out what we mean by this
+tag - for example if the journal stored the events as json it may try to find those with the field ``tag`` set to this value etc.).
+
+.. includecode:: code/docs/persistence/query/PersistenceQueryDocSpec.scala#events-by-tag
+
+As you can see, we can use all the usual stream combinators available from `Akka Streams`_ on the resulting query stream,
+including for example taking the first 10 and cancelling the stream. It is worth pointing out that the built-in ``EventsByTag``
+query has an optionally supported offset parameter (of type ``Long``) which the journals can use to implement resumable-streams.
+For example a journal may be able to use a WHERE clause to begin the read starting from a specific row, or in a datastore
+that is able to order events by insertion time it could treat the Long as a timestamp and select only older events.
+Again, specific sapabilities are specific to the journal you are using, so you have to
+
```

Comment:
I'll add that in a separate PR - have a ticket open for "query side and event adapters"